### ISSUE TITLE
Template Substitions are not supported for YAML

### ISSUE THREAD

**Issue by karadjov** [aws/aws-lambda-dotnet#220]
Currently Template Substitions are supported only for JSON.
https://github.com/aws/aws-lambda-dotnet/blob/911103991bb6c8b22216dc419f9cfb6986eaa764/Libraries/src/Amazon.Lambda.Tools/Utilities.cs#L450
This means we cannot use YAML if we want to use lambda **deploy-serverless** or **lambda package-ci**.

**Comment by normj** [comment]
Yes, I haven't implemented YAML substitution yet. Its on my list of things to do, a pull request is also welcome :)
